packages/manager/modules/pci/src/projects/project/storages/databases/databases.controller.js

```diff
@@ -40,7 +47,30 @@ export default class {
 
   deleteDatabase(database) {
     this.trackDatabases(`${optionsMenuTrackPrefix}delete_database`);
-    this.goToDeleteDatabase(database);
+    if (
+      [DATABASE_TYPES.KAFKA, DATABASE_TYPES.KAFKA_MIRROR_MAKER].includes(
+        database.engine,
+      )
+    ) {
+      this.DatabaseService.getIntegrations(
+        this.projectId,
+        database.engine,
+        database.id,
+      ).then((integrations) => {
+        const linkedServices = integrations.map((integration) =>
+          database.engine === DATABASE_TYPES.KAFKA
+            ? find(this.databases, { id: integration.destinationServiceId })
+            : find(this.databases, { id: integration.sourceServiceId }),
+        );
+        if (linkedServices.length > 0) {
+          this.goToConfirmDeleteDatabase(database, linkedServices);
+        } else {
+          this.goToDeleteDatabase(database);
+        }
+      });
+    } else {
+      this.goToDeleteDatabase(database);
+    }
```

Comment:
As all those looks to be promises maybe we could `return`

Reply:
Done here: https://github.com/ovh/manager/pull/6016/commits/18691ff34f865f6983a4e5322634bb08041bc977